packages/garbo-lib/src/wanderer/index.ts

```diff
@@ -50,22 +58,82 @@ function sober(): boolean {
 
 const wanderFactories: WandererFactory[] = [
   defaultFactory,
-  freefightFactory,
+  itemDropFactory,
   lovebugsFactory,
   guzzlrFactory,
   eightbitFactory,
   gingerbreadFactory,
   ultraRareFactory,
+  cookbookbatQuestFactory,
+  bofaFactory,
 ];
 
+function zoneAverageMonsterValue(
+  location: Location,
+  monsterValues: Map<Monster, number>,
+): number {
+  const rates = appearanceRates(location);
+  return sum([...monsterValues.keys()], (m) => {
+    const monsterValue = monsterValues.get(m) ?? 0;
+    const rate = rates[m.name] / 100;
+    return monsterValue * rate;
+  });
+}
+
+function targetedMonsterValue(
+  monsterValues: Map<Monster, number>,
+): [Monster, number] {
+  const bestMonster = maxBy(
+    [...monsterValues.keys()],
+    (m) => monsterValues.get(m) ?? 0,
+  );
+  return [bestMonster, monsterValues.get(bestMonster) ?? 0];
+}
+
+function initializeWandererLocationMap(
+  zoneValues: Map<
+    Location,
+    {
+      location: Location;
+      targets: WandererTarget[];
+      value: number;
+    }
+  >,
+): Map<Location, WandererLocation> {
+  // Build initial zones
+  const wandererLocationMap = new Map<Location, WandererLocation>();
+  for (const [locationKey, { location, targets, value }] of zoneValues) {
+    const wandererLocation: WandererLocation = {
+      location,
+      targets,
+      value,
+      peridotMonster: $monster`none`,
+    };
+    wandererLocationMap.set(locationKey, wandererLocation);
+  }
+  return wandererLocationMap;
+}
+
 function bestWander(
   type: DraggableFight,
   locationSkiplist: Location[],
   nameSkiplist: string[],
   options: WandererFactoryOptions,
 ): WandererLocation {
-  const possibleLocations = new Map<Location, WandererLocation>();
+  const locationZoneValues = new Map<
+    Location,
+    { location: Location; targets: WandererTarget[]; value: number }
+  >();
+  const locationMonsterValues = new Map<
+    Location,
+    {
+      location: Location;
+      targets: WandererTarget[];
+      monsterValues: Map<Monster, number>;
+    }
+  >();
```

Comment:
These two maps carry nearly identical information; why are they separate? why not have both a `value` and a `monsterValues` entry?